aips/aip-resource-access.md

```diff
@@ -0,0 +1,196 @@
+
+### Goals
+
+This AIP intends to achieve the following:
+
+- Have an organic extension of the Move language to support access control, which is fully downwards compatible
+- Enable new strategies for parallelization and sharding
+- Increase confidence in the effects of transactions and functions for users and auditors by access control declarations
+- Enable safe dynamic dispatch by making access control part of a type.
+
+### Out of Scope
+
+This feature will only be supported in the upcoming Aptos Move compiler ("compiler v2").
+
+Static analysis of access control will initially not be implemented, but is expected to be ready when AMC gets out of beta.
+
+## Motivation
+
+A Move function can read, write, and create arbitrary resources as long as it has access to the module APIs which allow this, and/or is in possession of a signer. There are multiple problems with this:
+
+- It has been identified in the past as a security risk. For example, a transaction that obtains a
+  signer and then delegates to some other well-known contract -- a 'helper' wrapper offered to some user -- can create arbitrary resources, including deploying code, without the user's knowledge, under the users account.
+- Unspecified read/write behavior of transactions makes it harder to parallelize for block execution and to implement sharding strategies.
+- It is difficult to reason about side effects of functions and transactions. This is both the case for auditors as for formal analysis like the Move prover. The only way to reason is to look at the code.
+- Since looking at code is the only way to reason about effects, _calling unknown_ code (dynamic dispatch across trust boundaries) is unsafe, and therefore currently forbidden in Move.
+
+## Impact
+
+Significant impact is expected in the medium term on two frontiers: on the one hand sharding and parallelization, on the other enabling other new language features like dynamic dispatch. 
+
+
+## Alternative solutions
+
+In Solana, a transaction must declare upfront the accounts it reads or writes from. Similarily, in Sui Move a transaction declares what objects it accesses. Using Aptos Objects as basis for what is proposed here has been discussed. Arguably, the approach described here is closer to the core Move language, as it does not depend on library extensions like objects. The approach is also more general, as it allows a variety of different access patterns, including negation.
+
+
+## Specification
+
+### Source Language
+
+The existing `acquires T` notation in Move is replaced by an extended syntax:
+
+```move
+fun f() acquires R {}                      // today: reads or write at any address; only local to module
+fun f() reads M1::R writes M2::T {}        // new access kinds -- note reference to types from other modules
+fun f<T> acquires R<T> {}                  // type instantiations supported
+fun f() acquires 0x42::*                   // wildcards
```

Comment:
You should include a body here and in the next line, or this is confusing.  Might be useful to bold-face the access specifier in the example to distinguish it from the other function spec syntax ("fun f() ... {}").